### ISSUE TITLE
Hi! I have the same error but with two different trunsactions:

### ISSUE THREAD

**Issue by rdagumampan** [rdagumampan/yuniql#205]
Hi! I have the same error but with two different trunsactions:

1. in ```/v0.00/crm/_crm_dev/``` I have:

```sql
-- v0.00
CREATE TABLE "FilterOperators" (
	"Id" uuid NOT NULL,
	"DisplayValue" text NULL,
	"Operation" text NULL,
	"SupportsList" bool NOT NULL,
	"Order" int4 NOT NULL DEFAULT 0,
	CONSTRAINT "PK_FilterOperators" PRIMARY KEY ("Id")
);
```

so I run: ``` yuniql.exe run --path "." --platform "postgresql" --connection-string "User ID=postgres;Password=changeme;Host=localhost;Port=5432;Database=crmdev;Pooling=true;" --auto-create-db --environment "crm_dev" --meta-table "_schema_version" --debug --bulk-separator "," --target-version "v0.00"```

Then
2. in ```v0.01/crm/_crm_dev/``` I have:

```csv
"Id","DisplayValue","Operation","SupportsList","Order"
"e520a9e3-c5c3-491d-bd95-ddebe1ff2237","Или","OR","false","0"
"0a03801e-710b-42e8-b65f-cf8ac7c81cd9","Не Выставлено","IS NULL","false","0"
"bb7650bd-56cd-4709-8a07-2a7573385ba2","Выставлено","IS NOT NULL","false","0"
"4d763591-09c2-4607-baac-deb5ce8eddc4","Выставлено","IS NOT NULL","false","0"
"8637fe30-270b-4641-a4f5-2c62ebe43b97","И","AND","false","0"
"11a5b14c-2c34-43e1-9ebe-5509823346af","Равно","IS NOT DISTINCT FROM","true","2"
"20bd53c2-3549-477a-b51e-fd7f13846e98","Не Содержит","NOT ILIKE","false","9"
"47d2062f-7066-46ff-aa04-cdcb896a39d0","Меньше","<","false","6"
"4d6f232e-2aef-40f1-9cfb-1366f925018d","Содержит","ILIKE","false","8"
"57cf19ed-8c16-4d56-be2a-36951c5bb75f","Больше",">","false","4"
"73b54e59-15f0-4b00-bd53-2ef20a82ec58","Меньше Или Равно","<=","false","7"
"d9d6b47c-0a65-407e-a551-6da355507a73","Не Равно","IS DISTINCT FROM","true","3"
"e8fc97a4-fd4f-4c00-ba81-3c800eaca899","Больше Или Равно",">=","false","5"
"f41d12d7-969b-404d-9cf4-5f844707cfba","Не Выставлено","IS NULL","false","1"

```

CSV File in UTF-8 Encoding.

so I run: ``` yuniql.exe run --path "." --platform "postgresql" --connection-string "User ID=postgres;Password=changeme;Host=localhost;Port=5432;Database=crmdev;Pooling=true;" --auto-create-db --environment "crm_dev" --meta-table "_schema_version" --debug --bulk-separator "," --target-version "v0.01"```

But it's failed with error:
```txt
...
INF   2021-05-12 14:14:01Z   Found 1 script files on .\v0.01\crm\_crm_debug_dev
  + FilterOperators.csv
INF   2021-05-12 14:14:01Z   PostgreSqlBulkImportService: Started copying data into destination table public.filteroperators
INF   2021-05-12 14:14:01Z   PostgreSqlBulkImportService: COPY public.filteroperators () FROM STDIN (FORMAT BINARY)
ERR   2021-05-12 14:14:01Z   Failed to execute run function. 42601: syntax error at or near ")"
Diagnostics stack trace captured a Npgsql.PostgresException (0x80004005):
   at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Npgsql.NpgsqlConnector.ReadMessage(DataRowLoadingMode dataRowLoadingMode)
   at Npgsql.NpgsqlBinaryImporter..ctor(NpgsqlConnector connector, String copyFromCommand)
   at Npgsql.NpgsqlConnection.BeginBinaryImport(String copyFromCommand)
   at Yuniql.PostgreSql.PostgreSqlBulkImportService.BulkCopyWithDataTable(IDbConnection connection, IDbTransaction transaction, String schemaName, String tableName, DataTable dataTable)
   at Yuniql.PostgreSql.PostgreSqlBulkImportService.Run(IDbConnection connection, IDbTransaction transaction, String fileFullPath, String bulkSeparator, Nullable`1 bulkBatchSize, Nullable`1 commandTimeout)
   at Yuniql.Core.MigrationService.<>c__DisplayClass17_0.<RunBulkImportScripts>b__1(String csvFile)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Yuniql.Core.MigrationService.RunBulkImportScripts(IDbConnection connection, IDbTransaction transaction, String workspace, String scriptDirectory, String bulkSeparator, Nullable`1 bulkBatchSize, Nullable`1 commandTimeout, String environment)
   at Yuniql.Core.MigrationService.<>c__DisplayClass12_1.<RunVersionDirectories>b__4(String scriptSubDirectory)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Yuniql.Core.MigrationService.<>c__DisplayClass12_0.<RunVersionDirectories>g__RunVersionDirectoriesInternal|3(IDbConnection connection, IDbTransaction transaction, List`1 scriptSubDirectories, String scriptDirectory, String versionDirectory, Stopwatch stopwatch)
   at Yuniql.Core.MigrationService.<>c__DisplayClass12_0.<RunVersionDirectories>b__2(String versionDirectory)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Yuniql.Core.MigrationService.RunVersionDirectories(IDbConnection connection, IDbTransaction transaction, List`1 appliedVersions, String workspace, String targetVersion, TransactionContext transactionContext, List`1 tokens, String bulkSeparator, String metaSchemaName, String metaTableName, Nullable`1 commandTimeout, Nullable`1 bulkBatchSize, String appliedByTool, String appliedByToolVersion, String environment, String transactionMode)
   at Yuniql.Core.MigrationService.<Run>g__RunAllInternal|11_3(IDbConnection connection, IDbTransaction transaction, Boolean isRequiredClearedDraft, <>c__DisplayClass11_0& )
   at Yuniql.Core.MigrationService.Run(String workspace, String targetVersion, Nullable`1 isAutoCreateDatabase, List`1 tokens, Nullable`1 isVerifyOnly, String bulkSeparator, String metaSchemaName, String metaTableName, Nullable`1 commandTimeout, Nullable`1 bulkBatchSize, String appliedByTool, String appliedByToolVersion, String environment, Nullable`1 isContinueAfterFailure, String transactionMode, Boolean isRequiredClearedDraft)
   at Yuniql.Core.MigrationService.Run()
   at Yuniql.CLI.CommandLineService.RunRunOption(RunOption opts)
  Exception data:
    Severity: ERROR
    SqlState: 42601
    MessageText: syntax error at or near ")"
    Position: 30
    File: scan.l
    Line: 1127
    Routine: scanner_yyerror
If you think this is a bug, please report an issue here https://github.com/rdagumampan/yuniql/issues.
```

I tested on windows 10 with:
```sh
 > yuniql.exe --version
yuniql 1.1.55
```

if instead of csv, I use sql in v0.01 It works.

```sql
INSERT INTO public."FilterOperators" ("Id","DisplayValue","Operation","SupportsList","Order") VALUES
	 ('e520a9e3-c5c3-491d-bd95-ddebe1ff2237'::uuid,'Или','OR',false,0),
	 ('0a03801e-710b-42e8-b65f-cf8ac7c81cd9'::uuid,'Не Выставлено','IS NULL',false,0),
	 ('bb7650bd-56cd-4709-8a07-2a7573385ba2'::uuid,'Выставлено','IS NOT NULL',false,0),
	 ('4d763591-09c2-4607-baac-deb5ce8eddc4'::uuid,'Выставлено','IS NOT NULL',false,0),
	 ('8637fe30-270b-4641-a4f5-2c62ebe43b97'::uuid,'И','AND',false,0),
	 ('11a5b14c-2c34-43e1-9ebe-5509823346af'::uuid,'Равно','IS NOT DISTINCT FROM',true,2),
	 ('20bd53c2-3549-477a-b51e-fd7f13846e98'::uuid,'Не Содержит','NOT ILIKE',false,9),
	 ('47d2062f-7066-46ff-aa04-cdcb896a39d0'::uuid,'Меньше','<',false,6),
	 ('4d6f232e-2aef-40f1-9cfb-1366f925018d'::uuid,'Содержит','ILIKE',false,8),
	 ('57cf19ed-8c16-4d56-be2a-36951c5bb75f'::uuid,'Больше','>',false,4),
	 ('73b54e59-15f0-4b00-bd53-2ef20a82ec58'::uuid,'Меньше Или Равно','<=',false,7),
	 ('d9d6b47c-0a65-407e-a551-6da355507a73'::uuid,'Не Равно','IS DISTINCT FROM',true,3),
	 ('e8fc97a4-fd4f-4c00-ba81-3c800eaca899'::uuid,'Больше Или Равно','>=',false,5),
	 ('f41d12d7-969b-404d-9cf4-5f844707cfba'::uuid,'Не Выставлено','IS NULL',false,1);
```
I think I have different error, but with the same (+/-) stacktrace.

_Originally posted by @zverev-iv in https://github.com/rdagumampan/yuniql/issues/129#issuecomment-839851066_

**Comment by rdagumampan** [comment]
@zverev-iv, thank you for reaching out and your interest with yuniql. I did quick look and I think its because you used Pascal case in the names of your table. Pgsql is quite sensitive to casing and I can see you double quoted the table names to preserve the casing but yuniql uses small case. Because yuniql uses small cased table name to extract table schema for destination of bulk import, it failed to get the destination schema columns. 

This logs shows where the problem is ` public.filteroperators`

```bash
INF   2021-05-12 14:14:01Z   PostgreSqlBulkImportService: Started copying data into destination table public.filteroperators
INF   2021-05-12 14:14:01Z   PostgreSqlBulkImportService: COPY public.filteroperators () FROM STDIN (FORMAT BINARY)
```

This should work:
 
 ```sql
-- v0.00
CREATE TABLE "filteroperators" (
	"Id" uuid NOT NULL,
	"DisplayValue" text NULL,
	"Operation" text NULL,
	"SupportsList" bool NOT NULL,
	"Order" int4 NOT NULL DEFAULT 0,
	CONSTRAINT "PK_FilterOperators" PRIMARY KEY ("Id")
);
```

But maybe this is not an option to you. Can you try to rename the CSV file into "FilterOperators".csv.

**Comment by zverev-iv** [comment]
@rdagumampan Thanks a lot!
Pascal Case in tables names is't my choise,  it's legacy option in project I currently working on.

I haven't written before (sorry, my mistake), but filename was ```v0.01/crm/_crm_dev/FilterOperators.csv```

but in logs we  see:
```log
INF   2021-05-13 13:41:14Z   Found 1 script files on .\v0.01\crm\_crm_dev
  + FilterOperators.csv
INF   2021-05-13 13:41:14Z   PostgreSqlBulkImportService: Started copying data into destination table public.filteroperators
INF   2021-05-13 13:41:14Z   PostgreSqlBulkImportService: COPY public.filteroperators () FROM STDIN (FORMAT BINARY)
ERR   2021-05-13 13:41:14Z   Failed to execute run function. 42601: syntax error at or near ")"
```

And ```"FilterOperators".csv``` name can't be used in windows/ntfs especially :(

**Comment by rdagumampan** [comment]
@zverev-iv , I have marked this now as a bug and I will prioritize this in the next release. A solution I think would be to detect if the name of CSV file has CAPITAL letters and preserve the name when used during bulk import. The only work around we have so far would be to loadup on scripts as you already did. Sorry for this inconvenience.

**Comment by rdagumampan** [comment]
Related #https://github.com/npgsql/npgsql/issues/3737

**Comment by rdagumampan** [comment]
@zverev-iv, a fixed is underway into master and I have verified this would enable Pascal names in CSV files. Thank you very much for this feedback. This fix will be released in v1.2 coming this summer.

![image](https://user-images.githubusercontent.com/5895952/118372584-b1febd00-b5b2-11eb-937d-b685501c6c24.png)

![image](https://user-images.githubusercontent.com/5895952/118372594-c347c980-b5b2-11eb-82c9-a6dadc83e786.png)

**Comment by rdagumampan** [comment]
Done. Will be released in v1.2.
https://github.com/rdagumampan/yuniql/commit/2c0b57e0da3b764affb478e9f0d483c2209895ff
